kratos/geometries/hexahedra_3d_20.h

```diff
@@ -878,11 +878,25 @@ template<class TPointType> class Hexahedra3D20 : public Geometry<TPointType>
      */
     void PrintData( std::ostream& rOStream ) const override
     {
+        // Base Geometry class PrintData call
         BaseType::PrintData( rOStream );
         std::cout << std::endl;
-        Matrix jacobian;
-        this->Jacobian( jacobian, PointType() );
-        rOStream << "    Jacobian in the origin\t : " << jacobian;
+
+        // Check if the geometry has valid points
+        bool is_valid = true;
+        for (IndexType i = 0; i < this->PointsNumber(); ++i) {
+            if (this->pGetPoint(i) == nullptr) {
+                is_valid = false;
+                break;
+            }
+        }
+
+        // If the geometry has valid points, calculate and output its data
+        if (is_valid) {
+            Matrix jacobian;
+            this->Jacobian( jacobian, PointType() );
+            rOStream << "    Jacobian in the origin\t : " << jacobian;
+        }
```

Comment:
how about adding a `protected` function `AllPointsValid` (perhaps you can come up with a better name :D), and use this here?

```suggestion

        // Check if the geometry has valid points
      if (!this->AllPointsAreValid() {
          return;
          }
        // If the geometry has valid points, calculate and output its data
        Matrix jacobian;
        this->Jacobian( jacobian, PointType() );
        rOStream << "    Jacobian in the origin\t : " << jacobian;
```

the early return makes the code also much more readable IMO

Reply:
That would be great (I also like the naming).

Reply:
Done.